src/main/java/org/rumbledb/runtime/flwor/expression/SimpleMapExpressionClosure.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Authors: Stefan Irimescu, Can Berker Cikis
+ *
+ */
+
+package org.rumbledb.runtime.flwor.expression;
+
+import org.apache.spark.api.java.function.FlatMapFunction;
+import org.rumbledb.api.Item;
+import org.rumbledb.context.DynamicContext;
+import org.rumbledb.runtime.RuntimeIterator;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class SimpleMapExpressionClosure implements FlatMapFunction<Item, Item> {
+
+
+    private static final long serialVersionUID = 1L;
+    private final RuntimeIterator rightIterator;
+    private final DynamicContext dynamicContext;
+
+    public SimpleMapExpressionClosure(RuntimeIterator rightIterator, DynamicContext dynamicContext) {
+        this.rightIterator = rightIterator;
+        this.dynamicContext = dynamicContext;
```

Comment:
this should read here

this.dynamicContext = new DynamicContext(dynamicContext)

that way, you can remove line 47, making it considerably faster as you only have one object creation per thread instead of per call.

Reply:
The speed-up given an input sequence of 10.000.000 looks to be about 1 second. I should also mention that I took the original logic from `PredicateIteratorClosure.java`, which looks like the following:

```
    @Override
    public Boolean call(Item v1) throws Exception {
        List<Item> currentItems = new ArrayList<>();
        currentItems.add(v1);
        DynamicContext dynamicContext = new DynamicContext(this.dynamicContext);
        dynamicContext.addVariableValue("$$", currentItems);

        this.expression.open(dynamicContext);
        boolean result = RuntimeIterator.getEffectiveBooleanValue(this.expression);
        this.expression.close();
        return result;

    }
```
Should we change the implementation of this as well to match your suggestions for this closure?